os/arch/arm/src/amebasmart/amebasmart_serial.c

```diff
@@ -1377,6 +1380,43 @@ int up_getc(void)
 	return ch;
 }
 
+/****************************************************************************
+ * Name: up_flush_console
+ *
+ * Description:
+ *    This function is used to ensure that all characters in the UART buffer
+ *    are transmitted.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+void up_flush_console(void)
+{
+#ifdef CONFIG_UART4_SERIAL_CONSOLE
+	uint16_t nbyte;
+	irqstate_t flags = enter_critical_section();
+
+	/* To avoid duplicated calling up_flush_console(). */
+	if (g_log_flush_running) {
+		return;
```

Comment:
we should have leave_critical_section here.

Reply:
Thank you, Updated!

Reply:
Since we have enter critical here, there won't be duplicate calling of up_flush_console(), right?

Reply:
This prevents the same cpu-thread from calling this function several times.